### PR TITLE
Docker: seahorn support

### PR DESCRIPTION
**Pull request by alastairreid**
These dockerfiles build SeaHorn and its dependencies.

Currently untested - all I know is that installation does not fail.

### CODE REVIEW

File: docker/seahorn/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+WORKDIR /opt
+
+ARG YICES_VERSION=2.6.2
+RUN curl --location https://yices.csl.sri.com/releases/${YICES_VERSION}/yices-${YICES_VERSION}-x86_64-pc-linux-gnu-static-gmp.tar.gz > yices.tgz \
+  && tar xf yices.tgz \
+  && rm yices.tgz
+
+ARG Z3_VERSION=4.8.7
+RUN curl --location https://github.com/Z3Prover/z3/releases/download/z3-${Z3_VERSION}/z3-${Z3_VERSION}-x64-ubuntu-16.04.zip > z3.zip \
```
Comment (fshaked) [inline docker/seahorn/Dockerfile]: Two copies of z3? One here, and another one in the z3 dockerfile?

**Review by fshaked — COMMENTED** [review]
Can you also add a checkout of https://github.com/yvizel/verify-c-common (those are scripts for running SH, don't need to build anything)?

File: docker/seahorn/Dockerfile (hunk 1 above)
Comment (alastairreid) [reply docker/seahorn/Dockerfile]: ok, deleting that

Comment (alastairreid) [reply docker/seahorn/Dockerfile]: verify-c-common is already being cloned - just before seahorn.

The pattern is: do all the safe, unlikely to fail actions first (eg cloning repos) before starting on builds, etc..
This works a bit better with the docker cache when steps start failing.

Comment (alastairreid) [reply docker/seahorn/Dockerfile]: ps For things like this, it is often simpler just to make the changes yourself and force-push them to this PR's branch.
In particular, if you need changes to how seahorn is built, you should either change this PR or accept this PR and create a PR on top of it - much faster than having me make the changes you need.

**Review by fshaked — APPROVED** [review]

**Pull request merged**